### PR TITLE
Adding support for security provider integration in cortex cmdlets.

### PR DESCRIPTION
**Pull request by nalinrajgupta**
## Description

Adding support for security provider integration in cortext. 
Customer can provide SecurityProviderName and Offic365LocalBreakoutCategory when creating/updating VirtualWan object. 
Customer can also provide/update EnbleInternetSecurity flag in VpnConnection and VirtualHubVnetConnection.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by nalinrajgupta** [comment]
Powershell document review: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/123

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/ResourceManager/Network/Commands.Network/help/Get-AzureRmVirtualWanSupportedSecurityProvider.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+## DESCRIPTION
+Get the supported security providers for the virtual wan.
+
+## EXAMPLES
+
+### Example 1
+```powershell
+PS C:\> New-AzureRmResourceGroup -Location "West US" -Name "testRG" 
+PS C:\> New-AzureRmVirtualWan -ResourceGroupName "testRG" -Name "myVirtualWAN" -Location "West US" -AllowBranchToBranchTraffic $true
+PS C:\> Get-AzureRmVirtualWanSupportedSecurityProvider -ResourceGroupName "testRG" -Name "myVirtualWAN"
```
Comment (maddieclayton) [inline src/ResourceManager/Network/Commands.Network/help/Get-AzureRmVirtualWanSupportedSecurityProvider.md]: Please add example output for this cmdlet

**Comment by maddieclayton** [comment]
@nalinrajgupta Please ping me when you are ready for me to take another look.

File: src/ResourceManager/Network/Commands.Network/ChangeLog.md [hunk 2]
```diff
@@ -19,6 +19,17 @@
 --->
 
 ## Current Release
+* Added cmdlets to integrate security provider in cortex
```
[the remark this answers is not in the document]
Comment (nalinrajgupta) [reply src/ResourceManager/Network/Commands.Network/ChangeLog.md]: I've added code for test. Ritwik has the AI to re-record the tests.

**Comment by nalinrajgupta** [comment]
@maddieclayton : I've resolved all comments, can you please take a look ?

**Comment by maddieclayton** [comment]
@nalinrajgupta Please ping me when you get this test recording in, and I'll take a final look.

**Comment by nalinrajgupta** [comment]
@maddieclayton : I've resolved all comments and recorded tests also. Can you please take a look ?

File: src/ResourceManager/Network/Commands.Network/Cortex/VirtualWan/GetAzureRmVirtualWanSupportedSecurityProviderCommand.cs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.Network.Models;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+    using Microsoft.Azure.Management.Network;
+
+    [Cmdlet(
+        VerbsCommon.Get,
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VirtualWanSupportedSecurityProvider",
+        SupportsShouldProcess = true,
```
Comment (maddieclayton) [inline src/ResourceManager/Network/Commands.Network/Cortex/VirtualWan/GetAzureRmVirtualWanSupportedSecurityProviderCommand.cs]: Please remove SupportShouldProcess = true here - you don't need this for any cmdlet that does not make a change to the server.

File: src/ResourceManager/Network/Commands.Network/help/Update-AzureRmVirtualHubVnetConnection.md [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+# Update-AzureRmVirtualHubVnetConnection
+
+## SYNOPSIS
+Updates an existing HubVirtualNetworkConnection.
+
+## SYNTAX
+
+### ByHubVirtualNetworkConnectionName (Default)
+```
+Update-AzureRmVirtualHubVnetConnection -ResourceGroupName <String> [-ParentResourceName <String>]
```
Comment (maddieclayton) [inline src/ResourceManager/Network/Commands.Network/help/Update-AzureRmVirtualHubVnetConnection.md]: Can you update this again? Looks like some parameters are optional here that should not be optional.

File: src/ResourceManager/Network/Commands.Network/Cortex/HubVnetConnection/UpdateAzureRmHubVirtualNetworkConnectionCommand.cs [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        [Alias("HubVirtualNetworkConnectionId")]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = CortexParameterSetNames.ByHubVirtualNetworkConnectionResourceId,
+            HelpMessage = "The resource id of the hubvirtualnetworkconnection resource to modify.")]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = true,
```
Comment (maddieclayton) [inline src/ResourceManager/Network/Commands.Network/Cortex/HubVnetConnection/UpdateAzureRmHubVirtualNetworkConnectionCommand.cs]: Why is this now mandatory?

**Comment by nalinrajgupta** [comment]
@maddieclayton : Can you please approve the PR ? All validations has passed now.

**Comment by maddieclayton** [comment]
@nalinrajgupta Can you regenerate the help files one more time? It looks like the Update-AzureRmVirtualHubVnetConnection parameter sets are wrong (InputObject is optional, ResourceId is optional), but it's correct in the actual cs file. That's my only remaining comment.

**Pull request merged**